playback/src/decoder/mod.rs

```diff
@@ -1,10 +1,24 @@
-use std::fmt;
+use thiserror::Error;
 
 mod lewton_decoder;
-pub use lewton_decoder::{VorbisDecoder, VorbisError};
+pub use lewton_decoder::VorbisDecoder;
 
 mod passthrough_decoder;
-pub use passthrough_decoder::{PassthroughDecoder, PassthroughError};
+pub use passthrough_decoder::PassthroughDecoder;
+
+#[derive(Error, Debug)]
+pub enum AudioError {
+    #[error("Lewton Decoder Error: {0}")]
+    LewtonDecoder(String),
+    #[error("Passthrough Decoder Error: {0}")]
+    PassthroughDecoder(String),
+    #[error("Decoder OggData Error: Can't return OggData on Samples")]
+    OggData,
+    #[error("Decoder Samples Error: Can't return Samples on OggData")]
+    Samples,
```

Comment:
Maybe extract these into a `AudioPacket` error? They do not seem consistent with the two error types above.

Reply:
Sure. I'll do that some time this weekend.

Reply:
Keeping with #820 `AudioPacketError` and `AudioPacketResult` and `AudioError` should really be `DecoderError` to match `DecoderResult` don't you think?  I kinda like the `WhateverResult`, `WhateverError` pairing.

Reply:
Yes, that does sound better.

Reply:
Done.